Mathlib/Data/List/TakeDrop.lean

```diff
@@ -151,6 +151,71 @@ theorem span_eq_takeWhile_dropWhile (l : List α) : span p l = (takeWhile p l, d
 
 end Filter
 
+section Splits
+
+/-! ### Generate splits of a list. -/
+
+/-- Pairs of prefixes and suffixes of a list. -/
+def splits (xs : List α) : List (List α × List α) :=
+  List.map xs.splitAt (range (xs.length + 1))
+
+@[simp]
+lemma splits_nil : (List.nil (α:=α)).splits= [([], [])] := by rfl
+
+@[simp]
+lemma splits_cons (x : α) (xs : List α) :
+    List.splits (x :: xs) =
+    ([], x :: xs) :: List.map (fun td ↦ (x :: td.1, td.2)) (List.splits xs) := by
+  simp [List.splits, List.range_succ_eq_map]
```

Comment:
I am less convinced that this should be tagged with `simp`. Note that tagging with "simp" means that the tactic `simp` will automatically apply this lemma to change LHS to RHS, but in this case the RHS seems to be significantly more complex than the LHS.

Reply:
removed simp tag